Source/Core/DolphinQt2/Config/Graphics/GraphicsWindow.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/Graphics/GraphicsWindow.h"
+#include "DolphinQt2/Config/Graphics/AdvancedWidget.h"
+#include "DolphinQt2/Config/Graphics/EnhancementsWidget.h"
+#include "DolphinQt2/Config/Graphics/GeneralWidget.h"
+#include "DolphinQt2/Config/Graphics/HacksWidget.h"
+
+#include <QDialogButtonBox>
+#include <QTabWidget>
+#include <QVBoxLayout>
+
+GraphicsWindow::GraphicsWindow(QWidget* parent, X11Utils::XRRConfiguration* xrr_config)
+    : QDialog(parent), m_xrr_config(xrr_config)
+{
+  CreateMainLayout();
+  ConnectWidgets();
+
+  setWindowTitle(tr("Graphics"));
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+}
+
+void GraphicsWindow::CreateMainLayout()
+{
+  auto* main_layout = new QVBoxLayout();
+  m_tab_widget = new QTabWidget();
+  m_button_box = new QDialogButtonBox(QDialogButtonBox::Ok);
+
+  main_layout->addWidget(m_tab_widget);
+  main_layout->addWidget(m_button_box);
+
+  auto* general_widget = new GeneralWidget(this, m_xrr_config);
```

Comment:
Widgets shouldn't require their parent to be anything more specific than a `QWidget*`, since that's intended solely for UI purposes. If something more specific needs to be passed in for non-UI purposes, that should be a separate argument.

In this case, couldn't you attach the signals here, rather than in the constructors? That would also simplify the widgets' internal logic by making it more "pure" (e.g. state changes pushed in, rather than pulled in).